sympy/matrices/expressions/matpow.py

```diff
@@ -56,14 +56,25 @@ def doit(self, **kwargs):
             args = self.args
         base = args[0]
         exp = args[1]
-        if isinstance(base, MatrixBase) and exp.is_number:
+        if exp.is_zero and base.is_square:
+            try:
```

Comment:
To solve the issue in the first place (for powers of both Identity and Zero Matrix), changes were made that required further adding of conditions. 
Also there were other examples that gave wrong results for power zero (some were not tested in the test cases), to deal with that I had to add try/catch to return the proper instance.